src/top/run/loss_util.py

```diff
@@ -0,0 +1,100 @@
+"""
+Util classes & methods for computing loss.
+Reference: https://github.com/xingyizhou/CenterNet/blob/master/src/lib/models/losses.py
+"""
+
+import torch as th
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+def _gather_feat(feat, ind, mask=None):
+    dim  = feat.size(2)
+    ind  = ind.unsqueeze(2).expand(ind.size(0), ind.size(1), dim)
+    feat = feat.gather(1, ind)
+    if mask is not None:
+        mask = mask.unsqueeze(2).expand_as(feat)
+        feat = feat[mask]
+        feat = feat.view(-1, dim)
+    return feat
+
+def _transpose_and_gather_feat(feat, ind):
+    feat = feat.permute(0, 2, 3, 1).contiguous()
+    feat = feat.view(feat.size(0), -1, feat.size(3))
+    feat = _gather_feat(feat, ind)
+    return feat
+
+
+def _neg_loss(pred, gt):
+    """
+    Focal loss same as CornerNet.
+    Args:
+        pred (batch x c x h x w)
```

Comment:
we should stick to either `(batch x channels x height x width)` or `(b x c x h x w)`. Personally I prefer `BCHW` or `(B,C,H,W)` since `x`s can be quite confusing. `NCHW` is also quite common.